pkg/client/unversioned/request.go

```diff
@@ -149,6 +150,10 @@ func (r *Request) Resource(resource string) *Request {
 		r.err = fmt.Errorf("resource already set to %q, cannot change to %q", r.resource, resource)
 		return r
 	}
+	if ok, msg := validation.ValidatePathSegmentName(resource, false); !ok {
```

Comment:
This is not an appropriate place to do this check. It should get moved to HPA code or server code.